src/Http/Controllers/CheckoutController.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Rapidez\Core\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+class CheckoutController
+{
+    public function __invoke(Request $request, ?string $step = null)
+    {
+        $checkoutSteps = config('rapidez.frontend.checkout_steps.' . config('rapidez.store_code')) ?: config('rapidez.frontend.checkout_steps.default');
+
+        if (! $step) {
+            $step = $checkoutSteps[0];
+        }
+
+        abort_if(! in_array($step, $checkoutSteps), 404);
+
+        // TODO: Is there a way to check if we should be able to go to this step?
+        // For example; without a quote we could be redirected to the homepage.
```

Comment:
Correct, the middleware will check if you have an active cart on every step.
However it will not check wether the order has the required info entered to go to the next step.

However the placeOrder graphql call will not permit placing the order if someone manually skips all steps.

Reply:
So the todo can be removed?